internal/bminventory/inventory.go

```diff
@@ -331,6 +343,52 @@ func (b *bareMetalInventory) getUserSshKey(params installer.GenerateClusterISOPa
 		"groups": [ "sudo" ]}`, sshKey)
 }
 
+func (b *bareMetalInventory) GetDiscoveryIgnition(ctx context.Context, params installer.GetDiscoveryIgnitionParams) middleware.Responder {
+	log := logutil.FromContext(ctx, b.log)
+
+	c, err := b.getCluster(ctx, params.ClusterID.String(), false)
+	if err != nil {
+		return common.GenerateErrorResponder(err)
+	}
+	isoParams := installer.GenerateClusterISOParams{ClusterID: params.ClusterID, ImageCreateParams: &models.ImageCreateParams{}}
+
+	cfg, err := b.formatIgnitionFile(c, isoParams, false)
+	if err != nil {
+		log.WithError(err).Error("Failed to format ignition config")
+		return common.GenerateErrorResponder(err)
+	}
+
+	configParams := models.DiscoveryIgnitionParams{Config: cfg}
+	return installer.NewGetDiscoveryIgnitionOK().WithPayload(&configParams)
+}
+
+func (b *bareMetalInventory) UpdateDiscoveryIgnition(ctx context.Context, params installer.UpdateDiscoveryIgnitionParams) middleware.Responder {
+	log := logutil.FromContext(ctx, b.log)
+
+	_, err := b.getCluster(ctx, params.ClusterID.String(), false)
+	if err != nil {
+		return common.GenerateErrorResponder(err)
+	}
+
+	_, report, err := ign_3_1.Parse([]byte(params.DiscoveryIgnitionParams.Config))
```

Comment:
IS it possible that params.DiscoveryIgnitionParams.Config is a valid ignition config but when we will call MergeIgnitionConfig the merge will fail?

Reply:
Perhaps we should make sure the merge is valid

Reply:
It will probably remove some shared code lines as well.

Reply:
`MergeIgnitionConfig` runs a validation here https://github.com/openshift/assisted-service/pull/310/files#diff-c7fd5af7de323c4ac8efd0331ede6f79R516

Reply:
Oh I think I misunderstood the concern here. You're asking if we should validate the merged config when someone submits an override ... maybe?

The [ignition source](https://github.com/coreos/ignition/blob/f6d4e9232e5b117000de3ec48e2f020900a90cf0/config/merge/merge.go#L26) says that you can merge two valid configs and get an invalid one, but I was never able to make that happen.

Reply:
Also I've been thinking more about this. The ISO create would actually fail in this case. That would give the user an opportunity to fix their patch before the install actually starts. This would also be a more accurate gauge of if the final ignition is invalid as we don't have some of the data when the user posts to this endpoint (some info is provided specifically to the generate iso endpoint, I think)